### PR TITLE
Photo caching

### PR DESCRIPTION
**Pull request by IONEGUY**

### CODE REVIEW

File: UsersGitHub/UsersGitHub.Android/CurrentVersion.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using Android.Content.PM;
+using UsersGitHub.Interfaces;
+
+[assembly: Xamarin.Forms.Dependency(typeof(UsersGitHub.Droid.CurrentVersion))]
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub.Android/CurrentVersion.cs]: this line is no needed as soon as you've registered service in prism container

File: UsersGitHub/UsersGitHub.Android/CurrentVersion.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace UsersGitHub.Droid
+{
+    public class CurrentVersion : IAppVersion
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub.Android/CurrentVersion.cs]: also please rename CurrentVersion to CurrentVersionService and IAppVersion to IAppVersionService

File: UsersGitHub/UsersGitHub.iOS/AppDelegate.cs [hunk 3]
```diff
@@ -23,9 +26,17 @@ public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsAppli
         public override bool FinishedLaunching(UIApplication app, NSDictionary options)
         {
             global::Xamarin.Forms.Forms.Init();
-            LoadApplication(new App());
+            LoadApplication(new App(new AndroidInitializer()));
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub.iOS/AppDelegate.cs]: o_O
you've created AndroidInitializer for iOS project?
please, rename

File: UsersGitHub/UsersGitHub.Android/CurrentVersion.cs (hunk 1 above)
Comment (YuliaLoyko) [reply UsersGitHub/UsersGitHub.Android/CurrentVersion.cs]: same for iOS version

File: UsersGitHub/UsersGitHub.Android/CurrentVersion.cs (hunk 2 above)
Comment (YuliaLoyko) [reply UsersGitHub/UsersGitHub.Android/CurrentVersion.cs]: same for iOS

File: UsersGitHub/UsersGitHub/Services/UserService.cs [hunk 4]
```diff
@@ -35,29 +44,31 @@ public class UserService
...
-                {
-                    Name = repos.Name
-                });
+                var repos = Mapper.Map<Repository>(reposDto);
+                repositories.Add(repos);
             }
             return repositories;
         }
+
+        private void InitMapper()
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub/Services/UserService.cs]: ?

File: UsersGitHub/UsersGitHub/ViewModels/BaseViewModel.cs [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 
 namespace UsersGitHub.ViewModels
 {
 	public class BaseViewModel : BindableBase
 	{
-	    public INavigationService NavigationService { get; set; }
-
         public BaseViewModel(INavigationService navigationService)
         {
-            NavigationService = navigationService;
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub/ViewModels/BaseViewModel.cs]: why is it removed?

Comment (YuliaLoyko) [reply UsersGitHub/UsersGitHub/ViewModels/BaseViewModel.cs]: how does it work without it?

File: UsersGitHub/UsersGitHub/ViewModels/BaseViewModel.cs [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
         }
 	}
 }
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub/ViewModels/BaseViewModel.cs]: also wha't going on with alignment in this file?

File: UsersGitHub/UsersGitHub/ViewModels/LoginPageViewModel.cs [hunk 7]
```diff
@@ -24,16 +28,21 @@ public string UserLogin
...
-        public LoginPageViewModel(INavigationService navigationService, IPageDialogService dialogService)
+        public LoginPageViewModel(INavigationService navigationService, 
+            IPageDialogService dialogService,
+            IUserService userService,
+            IInternetConnectionService internetConnectionService)
             : base(navigationService)
         {
+            this.userService = userService;
             this.dialogService = dialogService;
+            internetConnectionService.Init();
```
Comment (YuliaLoyko) [inline UsersGitHub/UsersGitHub/ViewModels/LoginPageViewModel.cs]: you should init this service in App instead of initializing in each ViewModel

**Review by YuliaLoyko — CHANGES_REQUESTED** [review]

File: UsersGitHub/UsersGitHub.iOS/AppDelegate.cs (hunk 3 above)
Comment (IONEGUY) [reply UsersGitHub/UsersGitHub.iOS/AppDelegate.cs]: )

**Review by YuliaLoyko — APPROVED** [review]

**Pull request merged**